### PR TITLE
Add SLIC_*_IF_ROOT convenience macros

### LINKED ISSUE #107: Add a SLIC_ERROR_IF_ROOT macro
I would be nice to have a root logging option with a conditional test.

### PR DESCRIPTION
**Pull request by joshessman-llnl**
Resolves #107

### CODE REVIEW

File: src/common/logger.hpp [hunk 1]
```diff
@@ -102,4 +102,32 @@ void flush();
 */
 #define SLIC_DEBUG_ROOT(rank, msg) SLIC_DEBUG_IF(rank == 0, msg)
 
+/*!
+*****************************************************************************
+* \brief Macro that logs given error message only on rank 0 if EXP is true.
+*****************************************************************************
+*/
+#define SLIC_ERROR_IF_ROOT(EXP, rank, msg) SLIC_ERROR_IF((EXP) && (rank == 0), msg)
```
Comment (jamiebramwell) [inline src/common/logger.hpp]: `SLIC_ERROR_ROOT_IF` may be better because this kind of sounds like it should print if the root. Any thoughts @white238 ?

Comment (white238) [reply src/common/logger.hpp]: I agree.

**Review by jamiebramwell — APPROVED** [review]

**Comment by joshessman-llnl** [comment]
LGTM

**Review by white238 — APPROVED** [review]

**Comment by joshessman-llnl** [comment]
LGTM

**Pull request merged**